### PR TITLE
[BD-46] OLD: added FormAutosuggest component

### PR DESCRIPTION
**Pull request by PKulkoRaccoonGang**
## Description

- take Organization dropdown implementation and create a standalone component in the Paragon repository.

[Figma](https://www.figma.com/file/eGmDp94FlqEr4iSqy1Uc1K/Paragon-Design-System?node-id=31030%3A85367)

### Deploy Preview

[FormAutosuggest component](https://deploy-preview-1541--paragon-openedx.netlify.app/components/form/form-autosuggest)

## Merge Checklist

* [x] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
* [x] Does your change adhere to the documented [style conventions](https://github.com/openedx/paragon/blob/master/docs/decisions/0012-css-styling-conventions)?
* [x] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
* [x] Were your changes tested using all available themes (see theme switcher in the header of the deploy preview, under the "Settings" icon)?
* [x] Were your changes tested in the `example` app?
* [x] Is there adequate test coverage for your changes?
* [x] Consider whether this change needs to reviewed/QA'ed for accessibility (a11y). If so, please add `wittjeff` and `adamstankiewicz` as reviewers on this PR.

## Post-merge Checklist

* [x] Verify your changes were released to [NPM](https://www.npmjs.com/package/@edx/paragon) at the expected version.
* [x] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [x] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @PKulkoRaccoonGang!


When this pull request is ready, tag your edX technical lead.

File: src/Form/FormAutosuggest.jsx [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+        errorMessage,
+      }));
+
+      setIsOpen(false);
+    }
+
+    setDisplayValue(e.target.value);
+  };
+
+  if (handleFocus) { handleFocus(); }
```
Comment (monteri) [inline src/Form/FormAutosuggest.jsx:204]: Why these `if (handleFocus) { handleFocus(); }` and `if (handleBlur) { handleBlur(); }` are used here?

File: src/Form/FormAutosuggest.jsx [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+  });
+
+  const handleExpandLess = (e) => {
+    if (isOpen) {
+      setState(prevState => ({
+        ...prevState,
+        dropDownItems: '',
+      }));
+    } else {
+      // eslint-disable-next-line no-use-before-define
```
[the remark this answers is not in the document]
Comment (PKulkoRaccoonGang) [reply src/Form/FormAutosuggest.jsx:35]: Thanks, corrected

File: src/Form/FormAutosuggest.jsx (hunk 1 above)
Comment (PKulkoRaccoonGang) [reply src/Form/FormAutosuggest.jsx]: Deleted, thanks

**Review by adamstankiewicz — COMMENTED** [review]
There's a few a11y implications here, I believe. Take a look at the behavior in the example our accessibility specialist provided in the #paragon-working-group Slack channel:
* https://github.com/alphagov/accessible-autocomplete
* https://alphagov.github.io/accessible-autocomplete/examples/

Note that it supports using the down/up arrow keys to change the selected value in the menu. It also supports closing the menu on the "Escape" key press.

I'd also recommend taking a look at the above example's aria roles and see how we might be able to replicate them to make this `Form.Autosuggest` bit more a11y-proof (e.g., `aria-autocomplete="list"`, `autocomplete="off"`, `role="listbox"`).

File: src/Form/FormAutosuggest.jsx [hunk 3]
```diff
@@ -0,0 +1,280 @@
+import React, { useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import { KeyboardArrowUp, KeyboardArrowDown } from '../../icons';
+import Icon from '../Icon';
+import { Form, IconButton, Spinner } from '../index';
+
+const FormAutosuggest = ({
+  value,
+  isLoading,
+  handleChange,
```
[the remark this answers is not in the document]
Comment (PKulkoRaccoonGang) [reply src/Form/FormAutosuggest.jsx:10]: corrected, thanks

**Comment by PKulkoRaccoonGang** [comment]
This pull request has moved to - https://github.com/openedx/paragon/pull/1586

**Comment by openedx-webhooks** [comment]
@PKulkoRaccoonGang Even though your pull request wasn’t merged, please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/paragon&entry.867055334=https://github.com/openedx/paragon/pull/1541&entry.1484655318=https://github.com/PKulkoRaccoonGang&entry.752974735=2022-08-11+20:27&entry.1917517419=2022-08-26+12:04&entry.2133058324=No) so we can improve your experience in the future.

**Pull request closed without merging**